lib/utilities/reportHandler.js

```diff
@@ -73,9 +73,46 @@ function getSystemStats() {
     };
 }
 
+function _crrRequest(backbeatMetrics, rDetails, site, log, cb) {
+    const details = JSON.parse(JSON.stringify(rDetails));
+    // Add `site` as we're not using Backbeat's request parser for the API's URI
+    details.site = site;
+    return backbeatMetrics.getAllMetrics(details, (err, res) => {
+        if (err) {
+            log.error('error occured when retrieving metrics', {
+                method: '_crrRequest',
+                error: err,
+            });
+            return cb(null, {});
+        }
+        const { completions, backlog, throughput } = res;
+        if (!completions || !backlog || !throughput) {
+            log.error('could not get metrics from backbeat', {
+                method: 'getCRRStats',
+            });
+            return cb(null, {});
+        }
+        const stats = {
+            completions: {
+                count: parseInt(completions.results.count, 10),
+                size: parseInt(completions.results.size, 10),
+            },
+            backlog: {
+                count: parseInt(backlog.results.count, 10),
+                size: parseInt(backlog.results.size, 10),
+            },
+            throughput: {
+                count: parseInt(throughput.results.count, 10),
+                size: parseInt(throughput.results.size, 10),
+            },
+        };
+        return cb(null, stats);
+    });
+}
+
 function getCRRStats(log, cb) {
     log.debug('getting CRR stats', { method: 'getCRRStats' });
-    const { replicationEndpoints, localCache: redis } = config;
```

Comment:
@rahulreddy can you confirm that it's expected that CRR stats are only found in the persistent redis/sentinel cluster?

Reply:
@jonathan-gramain could also weigh in